lib/shelley/src/Cardano/Wallet/Shelley.hs

```diff
@@ -229,15 +234,16 @@ serveWallet
                 icarusApi  <- apiLayer (newTransactionLayer proxy pm el ) nl
                 shelleyApi <- apiLayer (newTransactionLayer proxy pm el) nl
                 let spl = newStakePoolLayer (genesisParameters np) nl
-                startServer
-                    proxy
-                    socket
-                    randomApi
-                    icarusApi
-                    shelleyApi
-                    spl
-                    ntpClient
-                pure ExitSuccess
+                withPoolsMonitoring databaseDir (genesisParameters np) nl $ do
```

Comment:
Since `withPoolsMonitoring` hides the `DBLayer`, should we make it pass a `StakePoolLayer` to the action?

Reply:
I don't follow? The stake pool layer is independent of the pools monitoring?